twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -210,21 +210,22 @@ protected void doStartUp() {
 
     @Override
     protected void doShutDown() {
-      // Wait for sometime for the container to stop
-      // TODO: Use configurable value for stop time (TWILL-192)
-      int maxWaitSecs = Constants.APPLICATION_MAX_STOP_SECONDS - 30;
-      maxWaitSecs = maxWaitSecs < 15 ? 15 : maxWaitSecs;
+      // Wait for sometime for the container to stop. Subtracts 15 seconds from the timeout to accommodate for
+      // the message propagation delay.
+      long timeoutMillis = getTerminationTimeoutMillis(Constants.APPLICATION_MAX_STOP_SECONDS, TimeUnit.SECONDS);
+      timeoutMillis = timeoutMillis < 30L ? timeoutMillis : timeoutMillis - 15L;
+
       try {
-        if (Uninterruptibles.awaitUninterruptibly(shutdownLatch, maxWaitSecs, TimeUnit.SECONDS)) {
+        if (Uninterruptibles.awaitUninterruptibly(shutdownLatch, timeoutMillis, TimeUnit.SECONDS)) {
           return;
         }
       } catch (Exception e) {
         LOG.error("Got exception while trying to stop runnable {}, instance {}", runnable, instanceId, e);
       }
       // Container has not shutdown even after maxWaitSecs after sending stop message,
       // we'll need to kill the container
-      LOG.warn("Killing runnable {}, instance {} after waiting {} secs", runnable, instanceId, maxWaitSecs);
-      killAndWait(maxWaitSecs);
+      LOG.warn("Killing runnable {}, instance {} after waiting {} secs", runnable, instanceId, timeoutMillis);
+      killAndWait(timeoutMillis);
```

Comment:
shouldn't this be translated to seconds first?

Reply:
Changed to use seconds.